### ISSUE TITLE
Pretty printer not working for arrays of rank 0

### ISSUE THREAD

**Issue by Upabjojr** [sympy/sympy#15267]
```python
In [3]: a = Array([], shape=())

In [4]: a
Out[4]: ---------------------------------------------------------------------------

IndexError: list index out of range
```

Check the tensor printers as well.

**Comment by ayushmankoul** [comment]
Hello @Upabjojr I tried to reciprocate your code to understand the issue.It seems without specifying parameters in `shape` attribute it will take `1,1` as parameters whereas it looks that it is mandatory to pass parameters within `[]` in `array`.Please correct me if there is some other interpretation of the issue and are you working on this ?If you don't mind, I would like to give it a try.

**Comment by Upabjojr** [comment]
I'm not currently working. It just happened to me, so I opened the PR to remember fixing it. No exceptions should be thrown by `str( )`, so this is definitely an issue.

**Comment by ayushmankoul** [comment]
@Upabjojr Can you please specify the file in `sympy/tensor/array` where we need to look for source code of this issue ?

**Comment by Upabjojr** [comment]
Maybe the array constructor should raise an exception. If the shape is `()`, then one element is required.

**Comment by ayushmankoul** [comment]
Yes that can be the possible solution for this issue. @jksuom Can you please have a look on this issue ?

**Comment by jksuom** [comment]
The length of shape is the dimension of the array. A 1-dim. array is a list, a 2-dim. array is a list of lists, etc..., but I don't think that a 0-dim. array could be anything useful. Therefore I would raise an error if shape is `()`.

**Comment by jksuom** [comment]
That could be possible with sparse arrays.
